worker/jwtparser/jwt.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+// JWTParser is a parser responsible for parsing JWT tokens.
+type JWTParser struct {
+	cache      *jwk.Cache
+	httpClient *http.Client
+	refreshURL string
+}
+
+// NewParser creates a new JWT parser with a default http client.
+func NewParser(refreshURL string) *JWTParser {
+	httpClient := &http.Client{
```

Comment:
This should come as a dependency, otherwise we can't proxy it.

Reply:
I've gone halfway and made the client an input to the worker but not opted to create a client as an input to the manifold, opting instead to use a default client.
This, at least, makes it easier to test the worker and I've added some tests there.